R/core_loaders.R

```diff
@@ -0,0 +1,217 @@
+#' Load a `.csv` file to a data frame
+#'
+#' @param file Path to the file to load
+#' @param ... Additional arguments to pass to [readr::read_csv()]
+#'
+#' @return A `data.frame`
+#' @noRd
+load_csv <- function(file, ...) {
+  check_requires("Reading CSV files", "readr")
+  readr::read_csv(file, ...)
+}
+
+#' Load a `.tsv` file to a data frame
+#'
+#' @param file Path to the file to load
+#' @param ... Additional arguments to pass to [readr::read_tsv()]
+#'
+#' @return A `data.frame`
+#' @noRd
+load_tsv <- function(file, ...) {
+  check_requires("Reading TSV files", "readr")
+  readr::read_delim(file, ...)
+}
+
+#' Load an `.h5ad` file to `AnnData`
+#'
+#' @param file Path to the file to load
+#' @param ... Additional arguments to pass to [anndata::read_h5ad()]
+#'
+#' @return An `anndata::AnnDataR6` object
+#' @noRd
+load_h5ad <- function(file, ...) {
+  check_requires("Loading AnnData objects", "anndata")
+  anndata::read_h5ad(file, ...)
+}
+
+#' Load an `.zarr` file to `AnnData`
+#'
+#' @param file Path to the file to load
+#' @param ... Additional arguments to pass to `anndata::read_zarr()`
+#'
+#' @return The path to the file
+#' @noRd
+load_anndata_zarr <- function(file, ...) {
+  cli_warn("Loading AnnData Zarr files is not yet supported")
+  file
+}
+
+#' Load a Parquet file
+#'
+#' @param file Path to the file to load
+#' @param ... Additional arguments to pass to [nanoparquet::read_parquet()]
+#'
+#' @return A data frame
+#' @noRd
+load_parquet <- function(file) {
+  check_requires("Reading Parquet files", "nanoparquet")
+  nanoparquet::read_parquet(file)
+}
+
+#' Load an `.fcs` file to `AnnData`
+#'
+#' @param file Path to the file to load
+#' @param ... Additional arguments to pass to ...
+#'
+#' @return An `anndata::AnnDataR6` object
+#' @noRd
+load_fcs <- function(file, ...) {
+  cli_warn("Loading FCS files is not yet supported")
+  file
+}
+
+#' Load an `.h5mu` file to `MuData`
+#'
+#' @param file Path to the file to load
+#' @param ... Additional arguments to pass to `mudata::read_h5mu()`
+#'
+#' @return A `MuData` object
+#' @noRd
+load_h5mu <- function(file, ...) {
+  # Note: this is probably not the best solution
+  check_requires("Loading MuData objects", "reticulate")
+  mudata <- reticulate::import("mudata")
+
+  mudata$read_h5mu(file, ...)
+}
+
+#' Maybe display an HTML file
+#'
+#' If interactive mode is enabled, open the file in the default browser.
+#' Otherwise just return the path to the file.
+#'
+#' @param file Path to the file to load
+#' @param ... Additional arguments to pass to [utils::browseURL()]
+#'
+#' @return NULL if interactive mode is enabled, the path to the file otherwise
+#' @noRd
+load_html <- function(file, ...) {
+  if (is_knitr_notebook()) {
+    lines <- readLines(file)
+    return(knitr::raw_html(paste(lines, collapse = "\n")))
+  }
+
+  if (interactive()) {
+    check_requires("Opening HTML files", "utils")
```

Comment:
I think **{utils}** should always be installed so probably don't need this check

Reply:
Good point!